vscode-ext/src/extension.ts

```diff
@@ -177,52 +216,38 @@ class ConfigurationProvider implements vscode.DebugConfigurationProvider {
 			}
 		}
 
-		let libraryPath: string;
-		if (globalContext.extensionMode === vscode.ExtensionMode.Development) {
-			libraryPath = path.join(path.dirname(globalContext.extensionPath), "target", "debug");
+		let libraryPath: string, cliPath: string;
+
+		const mode = globalContext.extensionMode;
+		const extensionPath = globalContext.extensionPath;
+
+		if (mode === vscode.ExtensionMode.Development) {
+			const debugPath = path.join(path.dirname(extensionPath), "target", "debug");
+			libraryPath = debugPath;
+			cliPath = path.join(debugPath, "mirrord");
 		} else {
-			libraryPath = globalContext.extensionPath;
+			await extractMirrordLayer();
+			libraryPath = extensionPath;
+			cliPath = path.join(extensionPath, "mirrord");
 		}
+
 		let [environmentVariableName, libraryName] = LIBRARIES[os.platform()];
 		config.env ||= {};
 		config.env[environmentVariableName] = path.join(libraryPath, libraryName);
 
-		if (!await isTargetInFile()) { // If target wasn't specified in the config file, let user choose pod from dropdown
-			let podNamespace: string;
-			try {
-				podNamespace = await parseNamespace();
-			} catch (e: any) {
-				vscode.window.showErrorMessage('Failed to parse mirrord config file', e.message);
-				return;
-			}
-			let k8sApi = getK8sApi();
-			// Get pods from kubectl and let user select one to mirror
-			let pods: { response: any, body: V1PodList } = await k8sApi.listNamespacedPod(podNamespace);
-			let podNames = pods.body.items.map((pod) => pod.metadata!.name!);
-			let impersonatedPodName = '';
-			await vscode.window.showQuickPick(podNames, { placeHolder: 'Select pod to mirror' }).then(async podName => {
+		// If target wasn't specified in the config file, let user choose pod from dropdown			
+		if (!await isTargetInFile()) {
+			let targetNamespace = await parseNamespace();
+			let targets = targetNamespace != undefined ? await getMirrordTargets(cliPath, targetNamespace) : await getMirrordTargets(cliPath);
```

Comment:
why not `let targets = getMirrordTargets(cliPath, targetNamespace)` as `targetNamespace` will be undefined anyway?